config/v1/types_console.go

```diff
@@ -26,6 +29,29 @@ type Console struct {
 type ConsoleSpec struct {
 	// +optional
 	Authentication ConsoleAuthentication `json:"authentication"`
+	// route contains hostname and serving certificate for replacing the default hostname,
+	// under which is console available.
+	// If a custom route is specified, a new route will be created with the provided hostname.
+	// The default console route will be maintained so its hostname is reserved.
+	// If not specified, default route will used.
+	// +optional
+	Route CustomRoute `json:"route,omitempty"`
```

Comment:
Given https://github.com/openshift/api/blob/master/imageregistry/v1/types.go#L302 exists, would we have any reason to support `Routes []ConsoleConfigRoute json:"routes,omitempty"`?

Reply:
I'm inclined to go with `Routes: []ConsoleConfigRoute`, but document that we currently support only one.  If there is ever a desire to support more, we don't have a messy API. 

Our current RFE is for a single route, but we can be mindful of the future.

Reply:
We can't make it an array unless we honor all the values the user sets. There are problems with two routes. For instance, if I visit route A, login, then visit route B, the session cookie won't be there for route B. I'll be prompted to login again.

Unless we have a compelling reason for multiple routes, I'd suggest we only support one value.

Reply:
We have been talking with @benjaminapetersen regarding having and array but support only a single entry but @spadgett is right that it could cause misunderstanding.